cmd/metrics-prometheus-collector/cmd/cmd.go

```diff
@@ -179,6 +191,8 @@ func RunE(_ *cobra.Command, _ []string) error {
 
 	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 	defer stop()
+	metrics.stop = stop
+	metrics.ctx = ctx
```

Comment:
My instinct tells me you shouldn't be storing a context variable but I don't see an easier way to do this.